esp/services/ws_store/espstorelib/daliKVStore.cpp

```diff
@@ -402,7 +402,7 @@ bool CDALIKVStore::fetch(const char * storename, const char * ns, const char * k
         xpath.appendf("/%s", key);
         if(!storetree->hasProp(xpath.str()))
         {
-            throw makeStringExceptionV(ECLWATCH_INVALID_QUERY_KEY, "DALI Keystore fetch: invalid key '%s' detected!", key);
+            throw makeStringExceptionV(ECLWATCH_INVALID_QUERY_KEY, "DALI Keystore fetch: Could not find key '%s'!", key);
```

Comment:
This is a slightly picky comment, but exceptions should not be used for expected behaviour.  In this case it is expected that users will query they value of keys that do not exist - because the first time the value is used it will not be present.
Better would be to log at this point, return false and for the caller to make use of the return value..